src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,342 @@
+import Box, { BoxProps } from "@vertigis/web/ui/Box";
+import { FormElementProps, FormElementRegistration } from "@geocortex/workflow/runtime";
+import Input, { InputProps } from "@vertigis/web/ui/Input";
+import React, { useState, useEffect } from "react";
+import Table, { TableProps } from "@vertigis/web/ui/Table";
+import TableBody from "@vertigis/web/ui/TableBody";
+import TableCell, { TableCellProps } from "@vertigis/web/ui/TableCell";
+import TableFooter from "@vertigis/web/ui/TableFooter";
+import TableHead from "@vertigis/web/ui/TableHead";
+import TableRow from "@vertigis/web/ui/TableRow";
+import IconButton, { IconButtonProps } from "@vertigis/web/ui/IconButton";
+import Check from "@vertigis/web/ui/icons/Check";
+import Close from "@vertigis/web/ui/icons/Close";
+import Edit from "@vertigis/web/ui/icons/Edit";
+
+import { FormatOptions, format } from "@vertigis/arcgis-extensions/utilities/number";
+
+type FooterRow = Record<string, any>;
+type RowData = Record<string, any>;
+type SettableBoxProps = Pick<BoxProps, "maxHeight" | "maxWidth">;
+type SettableTableProps = Pick<TableProps, "size" | "stickyHeader">;
+
+interface Column {
+    name: string;
+    alias?: string;
+    align?: TableCellProps["align"];
+    columnCalculation?: "sum" | "average" | ((values: any) => any);
+    editable?: boolean;
+    type?: "text" | "number";
+    numberFormat?: FormatOptions;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<RowData[]>,
+    SettableBoxProps,
+    SettableTableProps {
+    cols: Column[];
+    rows: RowData[];
+    footerLabel?: string;
+    onMouseEnter?: (row: RowData) => void;
+    onMouseLeave?: (row: RowData) => void;
+    onRowEdit?: (row: RowData) => RowData;
```

Comment:
This should return `void`. The returned row isn't being used by anything.

Reply:
fixed